edgelet/edgelet-http/src/util/hyperwrap.rs

```diff
@@ -46,6 +49,20 @@ impl Config {
     }
 }
 
+fn uri_to_proxy(uri: Uri) -> Result<Proxy, Error> {
+    let url = Url::parse(&uri.to_string())?;
+    let mut proxy = Proxy::new(Intercept::All, uri);
+
+    if let Some(password) = url.password() {
```

Comment:
Isn't it possible to have a "username@hostname" as a URI?  Is this useful to manage that case?